### PR TITLE
angular: use dynamic import syntax

### LINKED ISSUE #9926: lazy-loading module: loadChildren dynamic import not working Angular 8
##### **Overview of the issue**

```
./src/main/webapp/app/app-routing.module.ts 14:40
Module parse failed: Unexpected token (14:40)
File was processed with these loaders:
 * ./node_modules/angular2-template-loader/index.js
 * ./node_modules/cache-loader/dist/cjs.js
 * ./node_modules/thread-loader/dist/cjs.js
 * ./node_modules/ts-loader/index.js
 * ./node_modules/angular-router-loader/src/index.js
 * ./node_modules/tslint-loader/index.js
You may need an additional loader to handle the result of these loaders.
|                 {
|                     path: 'admin',
>                     loadChildren: () => import('./admin/admin.module').then(m => m.AdminModule),
|                 },
|                 ...errorRoute,

```
**tsconfig.json**

```
{
  "compilerOptions": {
    "target": "es2017",
    "module": "esnext", 
    "moduleResolution": "node",
    "sourceMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "removeComments": false,
    "noImplicitAny": false,
    "skipLibCheck": true,
    "suppressImplicitAnyIndexErrors": true,
    "outDir": "target/classes/static/app",
    "lib": ["es7", "dom"],
    "typeRoots": ["node_modules/@types"],
    "baseUrl": "./",
    "paths": {
      "app/*": ["src/main/webapp/app/*"]
    },
    "importHelpers": true,
    "allowJs": true
  },
  "include": ["src/main/webapp/app", "src/test/javascript/"],
  "exclude": ["node_modules"]
}

```

**app-routing.module.ts**
```

import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { errorRoute } from './layouts';
import { DEBUG_INFO_ENABLED } from 'app/app.constants';

@NgModule({
  imports: [
    RouterModule.forRoot(
      [
        {
          path: 'admin',
          loadChildren: () => import('./admin/admin.module').then(m => m.AdminModule),
        },
        ...errorRoute,
      ],
      { enableTracing: DEBUG_INFO_ENABLED }
    ),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
```

### PR DESCRIPTION
**Pull request by wmarques**
Fix #9926 

Had to add the `acorn` dependency to force webpack to use the correct versions, thanks npm...

See there for more info on the issue: https://github.com/webpack/webpack/issues/8656

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/package.json.ejs [hunk 1]
```diff
@@ -75,6 +75,7 @@
     <%_ if (protractorTests) { _%>
     "@types/selenium-webdriver": "4.0.0",
     <%_ } _%>
+    "acorn": "^6.1.1",
```
Comment (antarus) [inline generators/client/templates/angular/package.json.ejs]: don't use `^` in version please

**Comment by wmarques** [comment]
@vishal423 I'll double check, I had the same issue but when I switched to `esnext` module in `tsconfig` the problem was resolved.

**Comment by wmarques** [comment]
@vishal423 I confirm it works for both JIT and AOT mode, just tested now :)

**Comment by vishal423** [comment]
thanks @wmarques

**Comment by maznag** [comment]
@wmarques not working for "acorn": "6.1.1" but it works well for "acorn": "6.2.0". Can you verify please ?

**Comment by wmarques** [comment]
@maznag for me it's working with 6.1.1...

**Comment by maznag** [comment]
@wmarques thanks ! for me it's working juste for 6.2.0

**Pull request merged**